modules/localization/msf/common/util/frame_transform.cc

```diff
@@ -0,0 +1,507 @@
+ *
+ * Determines the central meridian for the given UTM zone.
+ *
+ * Inputs:
+ *     zone - An integer value designating the UTM zone, range [1,60].
+ *
+ * Returns:
+ *   The central meridian for the given UTM zone, in radians, or zero
+ *   if the UTM zone parameter is outside the range [1,60].
+ *   Range of the central meridian is the radian equivalent of [-177,+177].
+ *
+ */
+inline double utm_central_meridian(int zone) {
+  return (-183.0 + (zone * 6.0)) * sins_deg_to_rad;
+}
+
+/*
+ * footpoint_latitude
+ *
+ * Computes the footpoint latitude for use in converting transverse
+ * Mercator coordinates to ellipsoidal coordinates.
+ *
+ * Reference: Hoffmann-Wellenhof, B., Lichtenegger, H., and Collins, J.,
+ *   GPS: Theory and Practice, 3rd ed.  New York: Springer-Verlag Wien, 1994.
+ *
+ * Inputs:
+ *   y - The UTM northing coordinate, in meters.
+ *
+ * Returns:
+ *   The footpoint latitude, in radians.
+ *
+ */
+double footpoint_latitude(double y) {
+  double y_ = 0.0;
+  double alpha_ = 0.0;
+  double beta_ = 0.0;
+  double gamma_ = 0.0;
+  double delta_ = 0.0;
+  double epsilon_ = 0.0;
+  double n = 0.0;
+  double result = 0.0;
+
+  /* Precalculate n (Eq. 10.18) */
+  n = (sm_a - sm_b) / (sm_a + sm_b);
+
+  /* Precalculate alpha_ (Eq. 10.22) */
+  /* (Same as alpha in Eq. 10.17) */
+  alpha_ = ((sm_a + sm_b) / 2.0) * (1 + (pow(n, 2.0) / 4)
+           + (pow(n, 4.0) / 64));
+
+  /* Precalculate y_ (Eq. 10.23) */
+  y_ = y / alpha_;
+
+  /* Precalculate beta_ (Eq. 10.22) */
+  beta_ = (3.0 * n / 2.0)
+          + (-27.0 * pow(n, 3.0) / 32.0) + (269.0 * pow(n, 5.0) / 512.0);
+
+  /* Precalculate gamma_ (Eq. 10.22) */
+  gamma_ = (21.0 * pow(n, 2.0) / 16.0) + (-55.0 * pow(n, 4.0) / 32.0);
+
+  /* Precalculate delta_ (Eq. 10.22) */
+  delta_ = (151.0 * pow(n, 3.0) / 96.0)   + (-417.0 * pow(n, 5.0) / 128.0);
+
+  /* Precalculate epsilon_ (Eq. 10.22) */
+  epsilon_ = (1097.0 * pow(n, 4.0) / 512.0);
+
+  /* Now calculate the sum of the series (Eq. 10.21) */
+  result = y_ + (beta_ * sin(2.0 * y_)) + (gamma_ * sin(4.0 * y_))
+           + (delta_ * sin(6.0 * y_)) + (epsilon_ * sin(8.0 * y_));
+
+  return result;
+}
+
+/*
+ * maplatlon_to_xy
+ *
+ * Converts a latitude/longitude pair to x and y coordinates in the
+ * Transverse Mercator projection.  Note that Transverse Mercator is not
+ * the same as UTM; a scale factor is required to convert between them.
+ *
+ * Reference: Hoffmann-Wellenhof, B., Lichtenegger, H., and Collins, J.,
+ * GPS: Theory and Practice, 3rd ed.  New York: Springer-Verlag Wien, 1994.
+ *
+ * Inputs:
+ *    phi - Latitude of the point, in radians.
+ *    lambda - Longitude of the point, in radians.
+ *    lambda0 - Longitude of the central meridian to be used, in radians.
+ *
+ * Outputs:
+ *    xy - A 2-element array containing the x and y coordinates
+ *         of the computed point.
+ *
+ * Returns:
+ *    The function does not return a value.
+ *
+ */
+void maplatlon_to_xy(double phi, double lambda,
```

Comment:
use const for input variables.

Reply:
all functions in this file changes to use const for input variables.